From f78feb4315c942a36be1816ac26b2c07f31a6a20 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Sun, 9 Sep 2018 17:49:30 +0800
Subject: [PATCH] change symbols in zh_CN level introduction text to workaround
 game hang

The text layout code in the game expects a long string to have spaces,
new lines or line breaks to break the line to suite the display size.
However, Chinese texts usually doesn't satisfy this, so they may hang
the code.

Change full-width Chinese symbols in level introduction texts to western
half-width ones add a space, to workaround this problem.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 CorsixTH/Lua/languages/simplified_chinese.lua | 170 +++++++++---------
 1 file changed, 85 insertions(+), 85 deletions(-)

diff --git a/CorsixTH/Lua/languages/simplified_chinese.lua b/CorsixTH/Lua/languages/simplified_chinese.lua
index e878a790..eb48e139 100644
--- a/CorsixTH/Lua/languages/simplified_chinese.lua
+++ b/CorsixTH/Lua/languages/simplified_chinese.lua
@@ -2872,107 +2872,107 @@ dynamic_info = {
 }
 introduction_texts = {
   level17 =
-    "最后的警告－时刻关注您的声望－这是真正吸引病人前来就诊的关键。" ..
-    "如果您没有杀死太多的病人，且使就诊病人保持快乐，则不必太担心声望！" ..
-    "决定权就在您的手中。祝您好运。",
+    "最后的警告 -- 时刻关注您的声望 -- 这是真正吸引病人前来就诊的关键. " ..
+    "如果您没有杀死太多的病人, 且使就诊病人保持快乐, 则不必太担心声望! " ..
+    "决定权就在您的手中. 祝您好运. ",
   level1 =
-    "欢迎来到您的第一座医院！//" ..
-    "首先要摆放接待台，建造一般诊断室，并雇佣一名接待员和一名医生。" ..
-    "随后就可以等待开张了。" ..
-    "建造精神病诊断治疗室并雇佣一名精神病医生是一个好主意。" ..
-    "药房和护士也是治疗病人所必需的。" ..
-    "建造一个充气机房间就可以治疗头部肿胀患者。" ..
-    "您需要治愈10个病人，并使声望保持在200以上。",
+    "欢迎来到您的第一座医院! //" ..
+    "首先要摆放接待台, 建造一般诊断室, 并雇佣一名接待员和一名医生. " ..
+    "随后就可以等待开张了. " ..
+    "建造精神病诊断治疗室并雇佣一名精神病医生是一个好主意. " ..
+    "药房和护士也是治疗病人所必需的. " ..
+    "建造一个充气机房间就可以治疗头部肿胀患者. " ..
+    "您需要治愈10个病人,并使声望保持在200以上. ",
   level9 =
-    "当填满了卫生部的银行帐户并为部长大人购买了一辆轿车后，您现在又要开始新的工作了。" ..
-    "在这里您要面对很多问题。" ..
-    "只有拥有足够经验充分的员工和房间，您才能够顺利过关。" ..
-    "医院价值需要达到$200,000，且银行帐户上要有$400,000。" ..
-    "如果无法达到上述要求，则无法胜利完成任务。",
+    "当填满了卫生部的银行帐户并为部长大人购买了一辆轿车后, 您现在又要开始新的工作了. " ..
+    "在这里您要面对很多问题. " ..
+    "只有拥有足够经验充分的员工和房间, 您才能够顺利过关. " ..
+    "医院价值需要达到$200,000, 且银行帐户上要有$400,000." ..
+    "如果无法达到上述要求, 则无法胜利完成任务. ",
   level2 =
-    "在该区域内还有一些其它的疾病。" ..
-    "建造医院从而可以治疗更多的病人，并应该考虑建造研究部门。" ..
-    "记住保持内部的清洁，从而使声望尽可能地高－您将遇到患有舌头松弛症的病人，因此需要建造舌头诊治房间。" ..
-    "您也可以建造心电图房间来帮助诊断疾病。" ..
-    "这些房间都需要经过研究才能够被建造。现在你可以购买其它土地扩展医院－使用城镇地图就可以购买土地。" ..
-    "目标是声望300，银行现金为10000，且治愈40个病人。",
+    "在该区域内还有一些其它的疾病. " ..
+    "建造医院从而可以治疗更多的病人, 并应该考虑建造研究部门. " ..
+    "记住保持内部的清洁, 从而使声望尽可能地高 -- 您将遇到患有舌头松弛症的病人, 因此需要建造舌头诊治房间. " ..
+    "您也可以建造心电图房间来帮助诊断疾病. " ..
+    "这些房间都需要经过研究才能够被建造. 现在你可以购买其它土地扩展医院 -- 使用城镇地图就可以购买土地. " ..
+    "目标是声望300, 银行现金为10000, 且治愈40个病人. ",
   level7 =
-    "在这里您将受到卫生部的密切监察，因此要在赚钱的同时，努力提高自己的声望。" ..
-    "我们无法处理太多的医疗事故－这对于医院的运营是十分不利的。" ..
-    "确认所有员工都很快乐，并确认已经购买了所有需要的仪器装备。" ..
-    "声望需要达到600，且银行里需要有$200,000。",
+    "在这里您将受到卫生部的密切监察, 因此要在赚钱的同时, 努力提高自己的声望. " ..
+    "我们无法处理太多的医疗事故 -- 这对于医院的运营是十分不利的. " ..
+    "确认所有员工都很快乐, 并确认已经购买了所有需要的仪器装备. " ..
+    "声望需要达到600, 且银行里需要有$200,000. ",
   level5 =
-    "医院将非常繁忙，处理各种各样的病人。" ..
-    "医生都是刚刚毕业的实习医生，因此需要建造一间培训室对他们进行培训，提高能力。" ..
-    "您只有3名专家可以帮助培训这些员工，因此一定要让专家快乐。" ..
-    "另外要注意的是，医院的位置不是很好。" ..
-    "经常会发生地震。" ..
-    "地震将对医院中的机器产生损坏，从而影响医院的运营。" ..
-    "使您声望达到400以上，现金达到$50,000。另外需要治愈200个病人。",
+    "医院将非常繁忙, 处理各种各样的病人. " ..
+    "医生都是刚刚毕业的实习医生, 因此需要建造一间培训室对他们进行培训, 提高能力. " ..
+    "您只有3名专家可以帮助培训这些员工, 因此一定要让专家快乐. " ..
+    "另外要注意的是, 医院的位置不是很好. " ..
+    "经常会发生地震. " ..
+    "地震将对医院中的机器产生损坏, 从而影响医院的运营. " ..
+    "使您声望达到400以上, 现金达到$50,000. 另外需要治愈200个病人. ",
   level4 =
-    "使所有的病人快乐，保持治疗的高效率并尽量降低死亡人数。" ..
-    "声望是十分重要的，因此尽量赢得更高的声望。" ..
-    "不要太多担心收入情况－当声望提高了，收入也自然会提高。" ..
-    "您需要培训您的医生，拓宽他们的能力。" ..
-    "这样他们就可以更好的为病人服务。" ..
-    "胜利条件是声望达到500以上。",
+    "使所有的病人快乐, 保持治疗的高效率并尽量降低死亡人数. " ..
+    "声望是十分重要的, 因此尽量赢得更高的声望. " ..
+    "不要太多担心收入情况--当声望提高了, 收入也自然会提高. " ..
+    "您需要培训您的医生, 拓宽他们的能力. " ..
+    "这样他们就可以更好的为病人服务. " ..
+    "胜利条件是声望达到500以上. ",
   level14 =
-    "这里还有一个挑战－一个充满惊奇的医院。" ..
-    "如果您能够成功完成这个任务，则您将成为所有胜利者中的佼佼者。" ..
-    "不要认为完成这个任务就象吃蛋糕一样，这将是您所遇到的最艰苦的工作。" ..
-    "祝您好运！",
+    "这里还有一个挑战 -- 一个充满惊奇的医院. " ..
+    "如果您能够成功完成这个任务, 则您将成为所有胜利者中的佼佼者. " ..
+    "不要认为完成这个任务就象吃蛋糕一样, 这将是您所遇到的最艰苦的工作. " ..
+    "祝您好运! ",
   level15 =
-    "好的，下面是管理医院的一些技巧。//" ..
-    "医生需要各种帮助来诊断病人。您可以" ..
-    "建造另一个诊断类房间，例如高级诊断室。",
+    "好的, 下面是管理医院的一些技巧. //" ..
+    "医生需要各种帮助来诊断病人. 您可以" ..
+    "建造另一个诊断类房间, 例如高级诊断室. ",
   level8 =
-    "需要您来建造一座高效的医院。" ..
-    "很多人都无事可做，因此需要适量的裁员以保持高效。" ..
-    "记住治愈病人是很重要的一件事情，但是您更要从中赚钱。" ..
-    "让恶心呕吐的病人靠近清洁工人。" ..
-    "需要赚取$300,000就可以过关。",
+    "需要您来建造一座高效的医院. " ..
+    "很多人都无事可做, 因此需要适量的裁员以保持高效. " ..
+    "记住治愈病人是很重要的一件事情, 但是您更要从中赚钱. " ..
+    "让恶心呕吐的病人靠近清洁工人. " ..
+    "需要赚取$300,000就可以过关. ",
   level13 =
-    "您的高超管理技能被特殊机密部门获知。" ..
-    "他们将向您提高特别奖金，因为他们有一座被老鼠困扰的医院需要有效管理。" ..
-    "您必须杀死尽可能多的老鼠，并让清洁工人打扫干净。" ..
-    "接受这个任务？",
+    "您的高超管理技能被特殊机密部门获知. " ..
+    "他们将向您提高特别奖金, 因为他们有一座被老鼠困扰的医院需要有效管理. " ..
+    "您必须杀死尽可能多的老鼠, 并让清洁工人打扫干净. " ..
+    "接受这个任务? ",
   level16 =
-    "当对病人完成诊断后，需要建造处理和治疗类房间完成对病人的治疗工作。可以从" ..
-    "建造药房开始。在药房中需要一名护士分配各种药品。",
+    "当对病人完成诊断后, 需要建造处理和治疗类房间完成对病人的治疗工作. 可以从" ..
+    "建造药房开始. 在药房中需要一名护士分配各种药品. ",
   level6 =
-    "使用您的所有知识来建造一个运行平稳的医院，从而可以赚取利润并处理任何问题。" ..
-    "您需要注意一点，医院周围的空气对细菌繁殖，传染病流行非常适宜。" ..
-    "如果您没有保持医院的清洁，则将面对传染病的流行。" ..
-    "赚取$150,000，并使医院价值超过$140,000。",
+    "使用您的所有知识来建造一个运行平稳的医院, 从而可以赚取利润并处理任何问题. " ..
+    "您需要注意一点, 医院周围的空气对细菌繁殖, 传染病流行非常适宜. " ..
+    "如果您没有保持医院的清洁, 则将面对传染病的流行. " ..
+    "赚取$150,000, 并使医院价值超过$140,000. ",
   level12 =
-    "您现在遇到了最大的挑战。" ..
-    "我们为您的成功感到由衷地高兴，卫生部为您准备了一项顶级工作；他们需要有人建造另一个超级医院，赚钱的同时获取较高的声望。" ..
-    "您可以购买任何需要的土地，治疗各种疾病赢得各种奖励。" ..
-    "动心了吗？" ..
-    "赚取$650,000，治愈750个病人，使声望达到800就可以胜利过关。",
+    "您现在遇到了最大的挑战. " ..
+    "我们为您的成功感到由衷地高兴, 卫生部为您准备了一项顶级工作; 他们需要有人建造另一个超级医院, 赚钱的同时获取较高的声望. " ..
+    "您可以购买任何需要的土地, 治疗各种疾病赢得各种奖励. " ..
+    "动心了吗? " ..
+    "赚取$650,000, 治愈750个病人, 使声望达到800就可以胜利过关. ",
   level3 =
-    "这次您将在一个富裕地区建造医院。" ..
-    "卫生部希望您能够在这里赚取更多的利润。" ..
-    "开始时您被要求获取好的声望，但是一旦医院步入正轨，就可以集中精力赚取更多的钱。" ..
-    "有可能会发生紧急事件。" ..
-    "也就是说一次会有大量病人就诊，且他们的病情都一样。" ..
-    "在时间限制内如果能够治愈他们，则不仅可以拿到奖金，声望也会提高。" ..
-    "一些疾病如猫王综合症等有可能发生，因此最好建造一间手术中心和附属病房。" ..
-    "胜利条件是赚取$20,000。",
+    "这次您将在一个富裕地区建造医院. " ..
+    "卫生部希望您能够在这里赚取更多的利润. " ..
+    "开始时您被要求获取好的声望, 但是一旦医院步入正轨, 就可以集中精力赚取更多的钱. " ..
+    "有可能会发生紧急事件. " ..
+    "也就是说一次会有大量病人就诊, 且他们的病情都一样. " ..
+    "在时间限制内如果能够治愈他们, 则不仅可以拿到奖金, 声望也会提高. " ..
+    "一些疾病如猫王综合症等有可能发生, 因此最好建造一间手术中心和附属病房. " ..
+    "胜利条件是赚取$20,000. ",
   level10 =
-    "随着您的经验不断增长，卫生部要求您集中精力提高药品的治疗效果。" ..
-    "有人对您颇有微辞，为此您必须使所有的药物都非常有效。" ..
-    "另外，一定要降低医疗事故的发生次数，减少死亡人数。" ..
-    "作为提示，您需要为建造胶桶留一些空地。" ..
-    "使所有药物的疗效都达到80%%，声望达到650且在银行帐户上有$500,000，这样就可以胜利过关。",
+    "随着您的经验不断增长, 卫生部要求您集中精力提高药品的治疗效果. " ..
+    "有人对您颇有微辞, 为此您必须使所有的药物都非常有效. " ..
+    "另外, 一定要降低医疗事故的发生次数, 减少死亡人数. " ..
+    "作为提示, 您需要为建造胶桶留一些空地. " ..
+    "使所有药物的疗效都达到80%%, 声望达到650且在银行帐户上有$500,000, 这样就可以胜利过关. ",
   level11 =
-    "现在您有机会建造一座终极医院。" ..
-    "该地区享有极高的声望，因此卫生部希望能够看到最好的医院建造在这里。" ..
-    "我们希望您能够赚取大量的金钱，获得很高的声望，并能够成功地处理任何事件。" ..
-    "这是一项非常重要的工作。" ..
-    "这需要您的努力工作。" ..
-    "注意一点，该区域常常会看到不明飞行物。因此请让您的员工做好准备迎接不速之客。" ..
-    "您的医院价值需要达到$240,000，在银行帐户内需要$500,000，且声望需要达到700。",
+    "现在您有机会建造一座终极医院. " ..
+    "该地区享有极高的声望, 因此卫生部希望能够看到最好的医院建造在这里. " ..
+    "我们希望您能够赚取大量的金钱, 获得很高的声望, 并能够成功地处理任何事件. " ..
+    "这是一项非常重要的工作. " ..
+    "这需要您的努力工作. " ..
+    "注意一点, 该区域常常会看到不明飞行物. 因此请让您的员工做好准备迎接不速之客. " ..
+    "您的医院价值需要达到$240,000, 在银行帐户内需要$500,000, 且声望需要达到700. ",
   level18 = "",
 }
 humanoid_name_starts = {
-- 
2.18.0

